src/dialects/mariadb/query-generator-typescript.ts

```diff
@@ -0,0 +1,39 @@
+import { rejectInvalidOptions } from '../../utils/check';
+import { joinSQLFragments } from '../../utils/join-sql-fragments';
+import { generateIndexName } from '../../utils/string';
+import { REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS } from '../abstract/query-generator-typescript';
+import type { RemoveIndexQueryOptions, TableNameOrModel } from '../abstract/query-generator-typescript';
+import { MySqlQueryGenerator } from '../mysql/query-generator.js';
+
+/**
+ * Temporary class to ease the TypeScript migration
+ */
+export class MariaDbQueryGeneratorTypeScript extends MySqlQueryGenerator {
+  removeIndexQuery(tableName: TableNameOrModel, indexNameOrAttributes: string | string[], options: RemoveIndexQueryOptions) {
+    if (options) {
+      rejectInvalidOptions(
+        'removeIndexQuery',
+        this.dialect.name,
+        REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS,
+        new Set(['ifExists']),
```

Comment:
```suggestion
        new Set<keyof RemoveIndexQueryOptions>(['ifExists']),
```
This way it checks if the options are keys in `RemoveIndexQueryOptions` so it would throw if you have a typo

Reply:
Now what I can't easily figure out yet is how to use this to check that `REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS` has all the keys of `RemoveIndexQueryOptions` and no others (or less). That way we still need to define it twice, but we check that the set is complete

Reply:
Add now to all dialects.
LMK if you have any questions